### ISSUE TITLE
Sync Multiple Videos on a same page

### ISSUE THREAD

**Issue by edoudou** [ampproject/amphtml#26630]
## Sync videos

Sometimes, especially on mobile devices in my experience, videos are not synced at all with each other. It could be great to make sure videos are synced.

### Alternative : Only sync videos on first play

I know how tricky it can get to keep videos sync at all time. Maybe just make sure they wait each other to start, and don't take care of the case were videos does not stay synced. 

### Warning : Compatibility

Make sure it work and the videos plays on every devices, as sometimes Apple's media player can react differently than the others.

**Comment by wassgha** [comment]
Hey @edoudou , could you add more information about your particular use case? What kind of syncing are you envisioning here?

**Comment by edoudou** [comment]
Yes no problem,

What I failed to mention was that I am in a context of using AMP-STORIES.
Here is a story were this feature would improve the result : https://recette.stories.teamjoin.fr/triple-riven-amp-S7qWfbP-l/
Sometimes, videos aren't synced as I explained earlier, especially on mobile devices.

**Comment by newmuis** [comment]
I think it's a reasonable feature request, but I also think that realistically we should encourage folks to explore doing something like this by creating a single video server-side.  I don't think all browsers will handle playing three videos simultaneously very well.

For example, because of the hardware decoding of videos on Apple devices, one or more of those three videos could fail to play on e.g. iOS Safari.  This is even true for a single video, but making it three videos makes it more likely to occur.

**Comment by edoudou** [comment]
It could be a good option, however it means the dimension would be fix, which mean you can have issues depending on your screen size. The video may act differently as expected on screens with a different ratio than yours.

Having multiple videos sent to the client and displayed by the web browser allow you to handle responsive more correctly.

**Comment by newmuis** [comment]
Yep, absolutely true.

I wonder if it's literally the same source whether multiple videos still take multiple decoders?  In theory we could expose APIs to allow multiple videos in layouts like this (or e.g. the [2-video background blur style](https://stackoverflow.com/questions/34675923/ffmpeg-how-to-convert-horizontal-video-169-to-vertical-video-916-with-blurre)) but ONLY allow it if the sources are the same (or, alternatively, expose APIs that take a single source and produce multiple videos with that source)

**Comment by newmuis** [comment]
This is a bit of a tangent; whether publishers specify the videos manually or we produce them internally to the framework, the original feature request to sync them still makes sense.  Let me split this off into a separate issue.

**Comment by gmajoulet** [comment]
> I think it's a reasonable feature request, but I also think that realistically we should encourage folks to explore doing something like this by creating a single video server-side.

+1 to this. I also understand that having three different elements is needed to create a responsive template, but maybe the solution would be to still have only one video with a fixed aspect ratio (letterboxing).
If we consider the average phone to be a ~$200 Android device, many devices won't be able to download/decode three videos at once, unless the video quality gets really, really low.

There are however things we can do to improve the video playback, like making sure all the videos are likely to play through without rebuffering before starting the playback simultaneously  (`readyState` to `HAVE_ENOUGH_DATA`). Waiting longer to ensure enough data is loaded to play with no rebuffer removes the need to sync the video as they run out of data to play, and would make the implementation much easier.
This might results in users waiting longer to see the videos start playing, depending on their device and bandwidth, but with that much content to load it is inevitable.

**Comment by newmuis** [comment]
I deleted my other comment, but I wonder if this is equally demanding performance-wise if the videos all have the same source?  Seems like in theory they'd only need to be downloaded and decoded once, but there'd be additional CPU overhead?

I feel like we should form an opinion here on the right way to do this, and ensure it as well as possible, since right now people can do whatever they'd like, leading to potentially broken experiences
